src/overlays/actors/ovl_Demo_Gj/z_demo_gj.c

```diff
@@ -96,241 +84,1326 @@ static ColliderCylinderInitType1 sCylinderInit = {
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[2]);
+            return;
+
+        case DemoGj_Type_22:
+            Collider_DestroyCylinder(globalCtx, &this->cylinders[0]);
+            return;
+
+        default:
+            break;
+    }
+}
+
+void DemoGj_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DemoGj* this = THIS;
+
+    DemoGj_Cylinder_Destroy(this, globalCtx);
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void DemoGj_PlayExplosionSfx(DemoGj* this, GlobalContext* globalCtx) {
+    Audio_PlaySoundAtPosition(globalCtx, &this->dyna.actor.world.pos, 50, NA_SE_EV_GRAVE_EXPLOSION);
+}
+
+static Vec3f sVelocity = { 0.0f, 6.0f, 0.0f };
+static Vec3f sAccel = { 0.0f, 0.0f, 0.0f };
+static Color_RGBA8 sPrimColor = { 0, 0, 0, 0 };
+static Color_RGBA8 sEnvColor = { 0, 0, 0, 0 };
+
+void func_80978AFC(GlobalContext* globalCtx, Vec3f* pos, f32 arg2) {
+    f32 temp = arg2 * 0.2f;
+    if (1) {
+        // this function just calls EffectSsDust_Spawn() with drawflags=1 and updateMode=0
+        func_800283D4(globalCtx, pos, &sVelocity, &sAccel, &sPrimColor, &sEnvColor, temp * Rand_ZeroOne() + arg2, 0xF,
+                      0x5A);
+    }
+}
+
+void DemoGj_DropCollectible(DemoGj* this, GlobalContext* globalCtx) {
+    Vec3f* pos = &this->dyna.actor.world.pos;
+    s16 collectible;
+    s32 amount;
+    s32 i;
+
+    collectible = DemoGj_GetCollectibleType(this);
+    amount = DemoGj_GetCollectibleAmount(this);
+
+    for (i = 0; i < amount; i++) {
+        Item_DropCollectible(globalCtx, pos, collectible);
+    }
+}
+
+void func_80978C20(DemoGj* this, GlobalContext* globalCtx, Vec3f* arg2, Vec3f* arg3) {
+    Vec3f pos;
+    Vec3f velocity;
+    s32 phi_s0;
+    f32 aux;
+    s16 theta = 0;
+    s32 i;
+
+    for (i = 0; i < 6; i++) {
+        pos.x = Math_SinS(theta) * 16.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = Math_CosS(theta) * 16.0f;
+
+        velocity.x = (pos.x * 0.6f) + (12.0f * arg3->x);
+        velocity.y = (Rand_ZeroOne() * 36.0f) + 6.0f;
+        velocity.z = (pos.z * 0.6f) + (12.0f * arg3->z);
+
+        pos.x += arg2->x;
+        pos.y += arg2->y;
+        pos.z += arg2->z;
+
+        aux = Rand_ZeroOne();
+        if (aux < 0.1f) {
+            phi_s0 = 0x61;
+        } else if (aux < 0.7f) {
+            phi_s0 = 0x41;
+        } else {
+            phi_s0 = 0x21;
+        }
+
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, arg2, -0xC8, phi_s0, 0xA, 0xA, 0,
+                             Rand_ZeroOne() * 20.0f + 20.0f, 0x14, 0x12C, (s32)(Rand_ZeroOne() * 30.0f) + 0x1E, -1,
+                             OBJECT_GEFF, &D_06000EA0);
+
+        theta += 0x2AAA;
+    }
+
+    DemoGj_PlayExplosionSfx(this, globalCtx);
+}
+
+s32 DemoGj_IsSceneInvalid(void) {
+    if (gSaveContext.sceneSetupIndex < 4) {
+        return 0; // return false; (?)
+    }
+    return 1; // return true; (?)
```

Comment:
same here, can use true and false.
about the name though, scene setup 4 and below will have some other kind of significance. Will need to reflect that moreso than just be called "valid". If you dont know what the significance is right now you can just leave the function unnamed